src/python/grpcio/grpc/_cython/_cygrpc/metadata.pyx.pxi

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+
+_Metadatum = collections.namedtuple('_Metadatum', ('key', 'value',))
+
+
+cdef void _c_metadata(metadata, size_t *c_count, grpc_metadata **c_metadata):
```

Comment:
Nit:  I'm used to seeing c functions that take an "output array of variable length" have the arguments ordered ```func(arg, char* buf, int* length)```

Reply:
Thanks for noticing; I think gRPC Core consistently uses that ordering as well. Changed both for this function and its inverse.